test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/concurrent/ConcurrentHashMapTest.scala

```diff
@@ -194,6 +194,34 @@ class ConcurrentHashMapTest extends MapTest {
         assertTrue(s"missing value '$value'", encounteredValues.contains(value))
     }
   }
+
+  @Test def addOnKeySetView(): Unit = {
+    val map = factory.empty[String, Int]
+    val keySet = map.keySet(0)
+    assertNull(map.get("ONE"))
+    assertTrue(keySet.add("ONE"))
+    assertEquals("Adding a new key adds the default value.", 0, map.get("ONE"))
+
+    assertFalse(keySet.add("ONE"))
+    assertEquals("Adding an existing key do not alter the value", 0, map.get("ONE"))
+
+    assertEquals(0, map.put("ONE", 42))
+    assertEquals("Putting alter the value", 42, map.get("ONE"))
+
+    assertFalse(keySet.add("ONE"))
+    assertEquals("Adding an existing key do not alter the value", 42, map.get("ONE"))
+  }
+
+  @Test def toStringOnKeySetView(): Unit = {
+    val map = factory.empty[String, Int]
+    val keySet = map.keySet(0)
+
+    map.put("a", 0)
+    assertEquals("[a]", keySet.toString)
+
+    map.put("b", 1)
+    assertEquals("[a, b]", keySet.toString)
```

Comment:
This test is likely overly restrictive: IIUC there is not guarantee of any particular key ordering when iterating over a `KeySetView`. So either `"[a, b]"` or `"[b, a]"` would be correct in this case.

Reply:
Fixed in https://github.com/scala-js/scala-js/pull/4367/commits/afbb9c57f285c8d60f66625178d97a4df1071d69